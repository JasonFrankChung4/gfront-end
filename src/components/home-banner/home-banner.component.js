/* global PUBLIC_PATH */

import React, { Fragment } from 'react'
import CustomLink from 'components/custom-link'
import HomeBannerSearchBox from '../home-banner-search-box'

export default class HomeBanner extends React.Component {

  componentDidMount () {
    if (!this.props.agencies) this.props.saveAgencies();
  }

  scrollToAbout() {
    const top = document.getElementById('banner-home').clientHeight;
    const offset = document.querySelector('header nav.main').clientHeight;
    window.scrollTo({
      top: top - offset,
      behavior: 'smooth'
    });
  }

  get agencyOptions() {
    if (this.props.agencies) {
      return this.props.agencies.map(agency => {
        return <option key={agency.acronym} value={agency.acronym}>{agency.name}</option>
      })
    } else {
      return null
    }
  }

  get browseDropdown() {
    if (!this.props.agencies) return null;

    return (
      <div className="browse">
        <span>or </span>
        <select onChange={this.props.onBrowseByEntityChange}>
          <option>{this.props.browseByText}</option>
          <option value="All">All</option>
          {this.agencyOptions}
        </select>
      </div>
    )
  }

  get verticalRow() {
    if (this.props.helpWantedTitle || this.props.helpWantedDescription) {
      return <div className="vertical-row" ></div>
    }
    return null
  }

  get helpWantedTitleSubsection() {
    if (this.props.helpWantedTitle) {
      return <div className="banner-subsection-title">{this.props.helpWantedTitle}</div>
    }
  }

  get helpWantedDescriptionSubsection() {
    if (this.props.helpWantedDescription) {
      return <div className="banner-subsection-subtitle">{this.props.helpWantedDescription}</div>
    }
    return null
  }

  get helpWantedButtonSubsection() {
    if (this.props.helpWantedButton) {
      return (
        <div className="buttons">
            <CustomLink to="/open-tasks">
              <button className="alt">{this.props.helpWantedButton}</button>
            </CustomLink>
        </div>
      )
    }
    return null
  }

  get issueSection() {
    if (this.props.issueUrl) {
      return (
        <div>
          <br/>
          <br/>
          <div className="banner-subsection">
            <div className="banner-subsection-subtitle" id="issue-banner-subsection-subtitle">
            <h2 className="banner-subsection-title">Connect with Us</h2>
              <img className="chat" src={PUBLIC_PATH + 'assets/img/icons/chat_bubble.png'}/>
              <span>Have questions or feedback? Open an issue on our <a className="home-link link" href={this.props.issueUrl} target="_blank">open source repository</a>.</span>
              <div>Want to receive our monthly newsletter? <a className="home-link link" href={this.props.newsletterUrl} target="_blank">Sign up here</a>.</div>
            </div>
          </div>
        </div>
      )
    }
    return null
  }

  render() {
    const sectionStyle = {
      backgroundImage: `url('${this.props.backgroundImage}')`
    }
    return (
      <section id="banner-home" className="banner large" style={sectionStyle}>
        <div className="banner-content">
          <div className="banner-title">{this.props.motto}</div>
          <div className="banner-subtitle show-w-gt-1200">{this.props.subtitle}</div>
          <div className="indented">
            <div className="banner-subsection width-half" style={{zIndex: 30}}>
              <div className="banner-subsection-content">
                <div className="banner-subsection-content-padder">
                  <HomeBannerSearchBox />
                  {this.browseDropdown}
                </div>
              </div>
            </div>

            {this.verticalRow}

            <div className="banner-subsection show-w-gt-600 width-half" id="banner-subsection-engage">
              <div className="banner-subsection-content">
                <div className="banner-subsection-content-padder">
                  {this.helpWantedTitleSubsection}
                  {this.helpWantedDescriptionSubsection}
                  {this.helpWantedButtonSubsection}
                </div>
              </div>
            </div>
          </div>
          {this.issueSection}
        </div>

        <a
          className="scroll-indicator"
          title="Scroll Down"
          onClick={this.scrollToAbout}
        >
          <i className="icon icon-angle-down"></i>
        </a>

      </section>
    )
  }
}
