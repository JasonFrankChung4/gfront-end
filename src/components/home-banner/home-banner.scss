@import '~@code.gov/code-gov-style/_sass/_variables';

section#banner-home {

  .banner-content {

    padding-top: 60px;

    /* adds some additional spacing between sections */
    .indented {
      @media screen and (max-width: 600px) {
        margin-top: 5rem;
      }
      @media screen and (min-height: 800px) and (min-width: 600px) {
        margin-top: 10rem;
      }
      @media screen and (max-height: 800px) and (min-width: 600px) {
        margin-top: 2rem;
      }
    }

    .banner-subtitle {
      @media screen and (max-width: 1200px) {
        display: none;
      }
    }

    .banner-subsection {
      height: 14rem;

      @media screen and (min-height: 800px) and (min-width: 600px) {
        height: 17rem;
      }

      .banner-subsection-content {
        @include center-vertically();

        .banner-subsection-content-padder {
          .buttons {
            padding-top: 7px;
          }
        }
      }

      @media screen and (max-width: 600px) {
        &#banner-subsection-engage {
          display: none;
        }
      }


      @media screen and (min-width: 1200px) {
        &:first-child {
          .banner-subsection-content-padder {
            padding-right: 80px;
          }
        }

        &:last-child {
          .banner-subsection-content-padder {
            padding-left: 80px;
          }
        }
      }
    }

    //div between Search and Help Wanted
    .vertical-row {
      @media screen and (min-width: 600px) {
        display: inline-block;
        height: 17rem;
        width: 3px;
        background: white;
        position: relative;
      }
    }
  }
  img.chat {
    height: 30px;
    vertical-align: middle;
  }
}