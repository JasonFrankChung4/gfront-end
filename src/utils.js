/*global ASSET_PATH */

import get from 'lodash.get'
import intersection from 'lodash.intersection'
import { run, startsWith } from '@code.gov/cautious'

export const falses = [undefined, null, 'null', 'None', 'Null', 'NULL', '', 'False', 'false']

export function isFalse(input) {
  return falses.includes(input)
}

export function isSet(input) {
  return input && typeof input === 'object' && input.has && input.add
}

export function adjustAssetPath(thing) {
  const pattern = /.?\/?assets\//
  const newAssetPath = ASSET_PATH + 'assets/'
  if (startsWith(thing, './assets') || startsWith(thing, '/assets/') || startsWith(thing, 'assets/')) {
    return thing.replace(pattern, newAssetPath)
  } else if (typeof value === 'object') {
    for (let key in thing) {
      const subvalue = thing[key]
      if (typeof subvalue === 'string') {
        thing[key] = adjustAssetPath(subvalue)
      }
    }
    return thing
  } else {
    return thing
  }
}

export function getConfigValue(siteConfig, path) {
  if (siteConfig) {
    let value = get(siteConfig, path)
    if (typeof value === 'string') {
      try {
        value = adjustAssetPath(value)
      } catch (error) {
        console.error("couldn't check if the following started with assets", [value])
      }
    } else if (Array.isArray(value)) {
      value = value.map(item => {
        if (typeof item === 'object') {
          for (let subkey in item) {
            let subvalue = item[subkey]
            if (typeof subvalue === 'string') {
              item[subkey] = adjustAssetPath(subvalue)
            }
          }
        }
        return item
      });
    }
    if (!value) {
      console.warn(`We weren't able to find the value for ${path} in your code-gov-config.json file.
      You can consult examples under the folder found at config/site/examples
      or consult our developer documentation here: https://developers.code.gov/configure.html`)
    }
    return value
  }
  else {
    return null
  }
}

export function normalize(input) {
  return trim(lower(input))
}

export function onEachItem(input, func) {
  if (Array.isArray(input)) {
    return input.map(item => run(item, func))
  } else if (isSet(input)) {
    return Set(Array.from(input).map(item => run(item, func)))
  } else {
    return run(input, func)
  }
}

export function lower(input) {
  return onEachItem(input, 'toLowerCase')
}

export function upper(input) {
  return onEachItem(input, 'toUpperCase')
}

export function trim(input) {
  return onEachItem(input, 'trim')
}

/* runs when each page component is loaded */
export function refreshView() {
  window.scrollTo(0, 0)
  document.activeElement.blur()
}


export function scrollToTopOfResults() {
  const headerHeight = document.querySelector("header.main").clientHeight
  const bannerHeight = document.querySelector(".banner").clientHeight
  const navHeight = document.querySelector("header.main nav").clientHeight
  const scrollDepth = headerHeight + bannerHeight - navHeight
  window.scrollTo(0, scrollDepth)
}

/* gets a set of values given a path */
export function getSet(items, path) {
  const results = new Set()
  if (Array.isArray(items) && typeof path === 'string') {
    items.forEach(item => {
      const value = get(item, path)
      if (value && value !== '') {
        if (Array.isArray(value)) {
          value.forEach(subitem => {
            if (isFalse(subitem) === false) {
              results.add(subitem)
            }
          })
        } else {
          if (isFalse(value) === false) {
            results.add(value)
          }
        }
      }
    })
  }
  return results
}

export function getLowerSet(items, path) {
  const results = new Set()
  if (Array.isArray(items) && typeof path === 'string') {
    items.forEach(item => {
      const value = get(item, path)
      if (value && value !== '') {
        if (Array.isArray(value)) {
          value.forEach(subitem => {
            if (isFalse(subitem) === false) {
              results.add(subitem.toLowerCase())
            }
          })
        } else {
          if(isFalse(value) === false) {
            results.add(value.toLowerCase())
          }
        }
      }
    })
  }
  return results
}

export function getFilterData(key, path, currentSearchResults, filters) {
  if (currentSearchResults && filters && filters[key]) {
    const names = getLowerSet(currentSearchResults.repos, path)
    return filters[key].filter(({ name, value }) => {
      return names.has(normalize(name)) || names.has(normalize(value))
    })
  }
}

export function hasLicense(repo) {
  return repo.permissions && Array.isArray(repo.permissions.licenses) && repo.permissions.licenses.length > 0
}

export function clean(string) {
  try {
    return string.toLowerCase().replace(/-/g,' ').trim()
  } catch (error) {
    return string
  }
}

export function sortByBestMatch(a, b) {
  return Number(a.searchScore) > Number(b.searchScore) ? -1 : 1
}

export function sortByName(a, b) {
  return clean(a.name) > clean(b.name) ? 1 : -1
}

export function getDate(repo, path) {

  if (repo.ghUpdatedAt) {
    return new Date(a.ghUpdatedAt).getTime()
  } else if (repo.date && repo.date.lastModified) {
    return new Date(a.date.lastModified).getTime()
  } else {
    return
  }
}

export function getTime(date) {
  try {
    if (date) {
      return new Date(date).getTime()
    } else {
      return -10e10
    }
  } catch (error) {
    return -10e10
  }
}

export function sortByDate(a, b) {
  try {
    const aDate = get(a, 'date.lastModified') || get(a, 'ghUpdatedAt')
    const bDate = get(b, 'date.lastModified') || get(b, 'ghUpdatedAt')
    const aTime = getTime(aDate)
    const bTime = getTime(bDate)
    return Math.sign(bTime - aTime) || 0;
  } catch (error) {
    return 0
  }
}

export function sortByDataQuality(a, b) {
  return Number(a.score) > Number(b.score) ? -1 : 1
}