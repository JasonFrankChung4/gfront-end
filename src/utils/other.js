/* global ASSET_PATH */
/* global URLSearchParams */

import get from 'lodash.get'
import intersection from 'lodash.intersection'
import { lower, run, startsWith, trim } from '@code.gov/cautious'
import siteConfig from '../../config/site/site.json'

export const falses = [undefined, null, 'null', 'None', 'Null', 'NULL', '', 'False', 'false']

export function isFalse(input) {
  return falses.includes(input)
}


export function adjustAssetPath(thing) {
  const pattern = /.?\/?assets\//
  const newAssetPath = ASSET_PATH + 'assets/'
  if (startsWith(thing, './assets') || startsWith(thing, '/assets/') || startsWith(thing, 'assets/')) {
    return thing.replace(pattern, newAssetPath)
  } else if (typeof value === 'object') {
    for (let key in thing) {
      const subvalue = thing[key]
      if (typeof subvalue === 'string') {
        thing[key] = adjustAssetPath(subvalue)
      }
    }
    return thing
  } else {
    return thing
  }
}

export function getConfigValue(path) {
  if (siteConfig) {
    let value = get(siteConfig, path)
    if (typeof value === 'string') {
      try {
        value = adjustAssetPath(value)
      } catch (error) {
        console.error("couldn't check if the following started with assets", [value])
      }
    } else if (Array.isArray(value)) {
      value = value.map(item => {
        if (typeof item === 'object') {
          for (let subkey in item) {
            let subvalue = item[subkey]
            if (typeof subvalue === 'string') {
              item[subkey] = adjustAssetPath(subvalue)
            }
          }
        }
        return item
      });
    }
    if (!value) {
      console.warn(`We weren't able to find the value for ${path} in your code-gov-config.json file.
      You can consult examples under the folder found at config/site/examples
      or consult our developer documentation here: https://developers.code.gov/configure.html`)
    }
    return value
  }
  else {
    return null
  }
}

export function normalize(input) {
  return trim(lower(input))
}

/* runs when each page component is loaded */
export function refreshView() {
  window.scrollTo(0, 0)
  document.activeElement.blur()
}


export function scrollToTopOfResults() {
  const headerHeight = document.querySelector("header.main").clientHeight
  const bannerHeight = document.querySelector(".banner").clientHeight
  const navHeight = document.querySelector("header.main nav").clientHeight
  const scrollDepth = headerHeight + bannerHeight - navHeight
  window.scrollTo(0, scrollDepth)
}

/* gets a set of values given a path */
export function getSet(items, path) {
  const results = new Set()
  if (Array.isArray(items) && typeof path === 'string') {
    items.forEach(item => {
      const value = get(item, path)
      if (value && value !== '') {
        if (Array.isArray(value)) {
          value.forEach(subitem => {
            if (isFalse(subitem) === false) {
              results.add(subitem)
            }
          })
        } else {
          if (isFalse(value) === false) {
            results.add(value)
          }
        }
      }
    })
  }
  return results
}

export function getLowerSet(items, path) {
  const results = new Set()
  if (Array.isArray(items) && typeof path === 'string') {
    items.forEach(item => {
      const value = get(item, path)
      if (value && value !== '') {
        if (Array.isArray(value)) {
          value.forEach(subitem => {
            if (isFalse(subitem) === false) {
              results.add(subitem.toLowerCase())
            }
          })
        } else {
          if(isFalse(value) === false) {
            results.add(value.toLowerCase())
          }
        }
      }
    })
  }
  return results
}

export function getFilterData(key, path, currentSearchResults, filters) {
  if (currentSearchResults && filters && filters[key]) {
    const names = getLowerSet(currentSearchResults.repos, path)
    return filters[key].filter(({ name, value }) => {
      return names.has(normalize(name)) || names.has(normalize(value))
    })
  }
}

export function hasLicense(repo) {
  return repo.permissions && Array.isArray(repo.permissions.licenses) && repo.permissions.licenses.length > 0
}


