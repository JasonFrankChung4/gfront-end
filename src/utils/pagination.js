import { last, range } from '@code.gov/cautious'

export function getPageInfo(params) {
  const { count, pagesize } = params
  const pagecount = Math.ceil(count / pagesize)
  const pageIndexes = range(pagecount)
    .map(n => n + 1) // convert from starting at 0 to 1
  const pageCount = pageIndexes.length

  return { pageIndexes, pageCount }
}

export function getPaginationNavInfo(params) {
  const { pageIndexes, page } = params
  const ultimate = last(pageIndexes)
  const left = page - 1
  const right = page + 1

  return { ultimate, left, right }
}
