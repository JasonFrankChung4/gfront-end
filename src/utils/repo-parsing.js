import get from 'lodash.get'
import { filter } from '@code.gov/cautious'

export function getDisplayTextForUsageType(repo) {
  let text = ''
  const usageType = get(repo, 'permissions.usageType')
  if (usageType === 'openSource') {
    text = 'Open Source'
  } else if (usageType === 'governmentWideReuse') {
    text = 'Gov-wide Reuse'
  } else if (usageType) {
    text = usageType
  } else {
    text = 'unknown'
  }
  return text
}

export function getLicenseName(repo) {
  return get(repo, 'permissions.licenses[0].name')
}

export function getLaborHours(repo) {
  try {
    const laborHours =  Number(get(repo, 'contact.laborHours'))
    if (laborHours > 0) {
      return laborHours
    }
  } catch (error){
    console.warn(error)
  }
}

export function parseLanguages(repo) {
  const languages = get(repo, 'languages')
  if (Array.isArray(languages)) {
    return filter(languages, Boolean)
  } else {
    return []
  }
}

export function parseEmail(repo) {
  const email = get(repo, 'contact.email')
  if (Boolean(email)) {
    return email
  }
}

export function parseRepositoryURL(repo) {
  const url = get(repo, 'repositoryURL')
  if (Boolean(url)) {
    return url
  }
}

export function parseTags(repo) {
  try {
    return get(repo, 'tags').filter(Boolean)
  } catch (error) {
    return []
  }
}


export function getLastModifiedDateString(repo) {
  try {
    const dateLastModified = get(repo, 'date.lastModified') ||  get(repo, 'ghUpdatedAt')
    if (dateLastModified) {
      return new Date(dateLastModified).toLocaleDateString()
    }
  } catch (error) {
    console.warn(error)
  }
}
