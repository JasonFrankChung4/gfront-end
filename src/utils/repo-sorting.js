export function clean(string) {
  try {
    return string.toLowerCase().replace(/-/g,' ').trim()
  } catch (error) {
    return string
  }
}

export function getDate(repo, path) {
  if (repo.ghUpdatedAt) {
    return new Date(repo.ghUpdatedAt).getTime()
  } else if (repo.date && repo.date.lastModified) {
    return new Date(repo.date.lastModified).getTime()
  } else {
    return
  }
}

export function getTime(date) {
  try {
    if (date) {
      return new Date(date).getTime()
    } else {
      return -10e10
    }
  } catch (error) {
    return -10e10
  }
}

export function sortByBestMatch(a, b) {
  return Number(a.searchScore) > Number(b.searchScore) ? -1 : 1
}

export function sortByDataQuality(a, b) {
  return Number(a.score) > Number(b.score) ? -1 : 1
}

export function sortByDate(a, b) {
  try {
    const aDate = get(a, 'date.lastModified') || get(a, 'ghUpdatedAt')
    const bDate = get(b, 'date.lastModified') || get(b, 'ghUpdatedAt')
    const aTime = getTime(aDate)
    const bTime = getTime(bDate)
    return Math.sign(bTime - aTime) || 0;
  } catch (error) {
    return 0
  }
}

export function sortByName(a, b) {
  return clean(a.name) > clean(b.name) ? 1 : -1
}
