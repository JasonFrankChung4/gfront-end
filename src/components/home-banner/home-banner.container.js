import { connect } from 'react-redux';
import { push } from 'connected-react-router'
import { getConfigValue, normalize } from 'utils/other'
import saveAgencies from 'actions/save-agencies'
import updateBrowseFilters from 'actions/update-browse-filters'
import HomeBannerComponent from './home-banner.component'

const mapStateToProps = ({ agencies, siteConfig }) => {
  return {
    agencies,
    backgroundImage: getConfigValue(siteConfig, 'images.background'),
    motto: getConfigValue(siteConfig, 'content.home.banner.motto'),
    subtitle: getConfigValue(siteConfig, 'content.home.banner.subtitle'),
    helpWantedTitle: getConfigValue(siteConfig, 'content.home.banner.help_wanted.title'),
    helpWantedDescription: getConfigValue(siteConfig, 'content.home.banner.help_wanted.description'),
    helpWantedButton: getConfigValue(siteConfig, 'content.home.banner.help_wanted.button'),
    issueUrl: getConfigValue(siteConfig, 'content.home.banner.issue_url'),
    searchDescriptionText: getConfigValue(siteConfig, 'content.home.banner.search_description_text'),
    searchDescriptionTextMobile: getConfigValue(siteConfig, 'content.home.banner.search_description_text_mobile'),
    browseByText: getConfigValue(siteConfig, 'content.home.banner.browse_by_text')
  }
}

const mapDispatchToProps = dispatch => {
  return {
    onBrowseByEntityChange: event => {
      const value = event.target.value
      if (value !== 'Browse by Agency') {
        let url = '/browse-projects'
        if (value !== 'All') {
          url += '?agencies=' + value
          dispatch(updateBrowseFilters('agencies', normalize(value)))
        }
        dispatch(push(url))
      }
    },
    saveAgencies: () => dispatch(saveAgencies())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(HomeBannerComponent)
