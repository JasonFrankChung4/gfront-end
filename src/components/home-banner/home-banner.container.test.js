import { push } from 'connected-react-router'

import saveAgencies from 'actions/save-agencies'
import updateBrowseParams from 'actions/update-browse-params'
import { now } from 'utils/other'
import { mapStateToProps, mapDispatchToProps } from 'components/home-banner/home-banner.container'

jest.mock('connected-react-router')
jest.mock('actions/save-agencies')
jest.mock('actions/update-browse-params')

const props = {
  agencies: [
    { acronym: 'a1', name: 'agency-1' },
    { acronym: 'a2', name: 'agency-2' },
  ],
}

const params = {
  page: 1,
  size: 10,
  sort: 'data_quality',
  filters: [],
}

const dispatch = jest.fn()

// helper function for testing `onBrowseByEntityChange` dispatches an action with expected params
const testByValue = ({ value, action, expected }) => {
  const event = { target: { value } }
  mapDispatchToProps(dispatch).onBrowseByEntityChange(event)
  expect(dispatch).toBeCalled()
  expect(action).toBeCalledWith(expected)
}

describe('containers - HomeBanner', () => {
  describe('mapStateToProps', () => {
    it('should return the correct properties', () => {
      expect(mapStateToProps(props)).toMatchSnapshot()
    })
  })

  describe('mapDispatchToProps', () => {
    describe('onBrowseByEntityChange', () => {
      describe('value is `browse by agency`', () => {
        it('should do nothing', () => {
          const event = { target: { value: 'browse by agency' } }
          mapDispatchToProps(dispatch).onBrowseByEntityChange(event)
          expect(dispatch).not.toBeCalled()
        })
      })

      describe('value is `all`', () => {
        it('should dispatch the `updateBrowseParams` action', () => {
          testByValue({ value: 'all', action: updateBrowseParams, expected: params })
        })

        it('should dispatch an action to push the url', () => {
          const expected = '/browse-projects?page=1&size=10&sort=data_quality'
          testByValue({ value: 'all', action: push, expected })
        })
      })

      describe('value is neither `all` nor `browse by agency`', () => {
        it('should dispatch the `updateBrowseParams` action with an agency filter', () => {
          const expected = {
            ...params,
            filters: [{ category: 'agencies', value: 'other-value', modified: now() }],
          }
          testByValue({ value: 'other-value', action: updateBrowseParams, expected })
        })

        it('should dispatch an action to push the url with the agency set', () => {
          const expected = `/browse-projects?agencies=other-value&page=1&size=10&sort=data_quality`
          testByValue({ value: 'other-value', action: push, expected })
        })
      })
    })

    describe('saveAgencies', () => {
      mapDispatchToProps(dispatch).saveAgencies()
      expect(dispatch).toBeCalled()
      expect(saveAgencies).toBeCalled()
    })
  })
})
