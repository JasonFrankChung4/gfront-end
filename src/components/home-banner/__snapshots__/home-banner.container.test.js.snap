// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers - HomeBanner mapStateToProps should return the correct properties 1`] = `
Object {
  "agencies": Array [
    Object {
      "acronym": "a1",
      "name": "agency-1",
    },
    Object {
      "acronym": "a2",
      "name": "agency-2",
    },
  ],
  "backgroundImage": "/assets/img/american_flag_1920x1280_gradient.jpg",
  "browseByText": "Browse by Agency",
  "helpWantedButton": "Explore Open Tasks",
  "helpWantedDescription": "Whether you are a beginner or an experienced coder, join the open source community. Help improve America's Code.",
  "helpWantedTitle": "Ready. Set. Code!",
  "issueUrl": "https://github.com/GSA/code-gov-front-end/issues",
  "motto": "Sharing America's Code",
  "newsletterUrl": "https://public.govdelivery.com/accounts/USGSATTS/subscriber/new?topic_id=USGSATTS_5",
  "subtitle": "Unlock the tremendous potential of the Federal Governmentâ€™s software.",
}
`
